server/src/main/java/org/opensearch/index/store/remote/utils/cache/LRUCache.java

```diff
@@ -45,15 +47,15 @@
     private final HashMap<K, Node<K, V>> data;
 
     /** the LRU list */
-    private final LinkedDeque<Node<K, V>> lru;
+    private final LinkedHashMap<K, Node<K, V>> lru;
 
     private final RemovalListener<K, V> listener;
 
     private final Weigher<V> weigher;
 
     private final StatsCounter<K> statsCounter;
 
-    private volatile ReentrantLock lock;
+    private final ReentrantLock lock;
```

Comment:
:+1:  we could replace

```
        final ReentrantLock lock = this.lock;
        lock.lock();
```

with just

```
        lock.lock();
```

now

Reply:
Oh good catch! I'm sure there was some previous incarnation of this code where a volatile lock was needed, but it definitely was never necessary here.